addons/payment/tests/test_account_journal.py

```diff
@@ -0,0 +1,17 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.exceptions import UserError
+from odoo.tests import tagged
+
+from odoo.addons.payment.tests.common import PaymentCommon
+
+
+@tagged('post_install', '-at_install')
+class TestAccountJournal(PaymentCommon):
+
+    def test_prevent_removing_apml_from_journal_of_active_provider(self):
+        """ Deleting an account.payment.method.line that is related to a provider in 'test' or 'enabled' state
+        should raise an error.
+        """
+        self.assertEqual(self.dummy_acquirer.state, 'test')
+        with self.assertRaises(UserError):
+            self.dummy_acquirer.journal_id.write({'inbound_payment_method_line_ids': [(5, 0, 0)]})
```

Comment:
Ideally, you should use the [`odoo.fields.Command` API](https://www.odoo.com/documentation/master/developer/reference/backend/orm.html#odoo.fields.Command.clear) instead of the magic numbers when working in >=15.0.